src/main/java/org/kitteh/irc/client/library/element/mode/Ban.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * * Copyright (C) 2013-2016 Matt Baxter http://kitteh.org
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.kitteh.irc.client.library.element.mode;
+
+import org.kitteh.irc.client.library.Client;
+import org.kitteh.irc.client.library.element.Channel;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+
+/**
+ * Represents a ban entry.
+ */
+public interface Ban {
+    @Nonnull
+    Optional<String> getBanner();
+
+    @Nonnull
+    Channel getChannel();
+
+    @Nonnull
+    Client getClient();
+
+    @Nonnull
+    String getId();
+
+    @Nonnull
+    Optional<Integer> getTimeLeft();
```

Comment:
It's not the time left, but the date the ban was set (as best known as by the IRCd. Which might not be very reliable to begin with)